tests/functional/conftest.py

```diff
@@ -205,25 +208,50 @@ def sync_wrap_low_wait_update(setup_patch: GenericPdsRequestor, create_random_da
 
 
 def set_quota_and_rate_limit(
-    product: ApigeeApiProducts,
+    apigeeObj: Union[ApigeeApiProducts, ApigeeApiDeveloperApps],
     rate_limit: str = "1000ps",
     quota: int = 60000,
     quota_interval: str = "1",
-    quota_time_unit: str = "minute"
+    quota_time_unit: str = "minute",
+    quota_enabled: bool = True,
+    rate_enabled: bool = True,
+    proxy: str = ""
 ) -> None:
-    """Sets the quota and rate limit on an apigee product.
+    """Sets the quota and rate limit on an apigee product or app.
 
     Args:
-        product (ApigeeApiProducts): Apigee product
+        obj (Union[ApigeeApiProducts,ApigeeApiDeveloperApps]): Apigee product or Apigee app
         rate_limit (str): The rate limit to be set.
         quota (int): The amount of requests per quota interval.
         quoata_interval (str): The length of a quota interval in quota units.
         quota_time_unit (str): The quota unit length e.g. minute.
+        quota_enabled (bool): Enable or disable proxy level quota.
+        rate_enabled (bool): Enable or disable proxy level spike arrest.
+        proxy (str): The proxy to apply rate limiting to.
     """
-    asyncio.run(product.update_ratelimits(quota=quota,
-                                          quota_interval=quota_interval,
-                                          quota_time_unit=quota_time_unit,
-                                          rate_limit=rate_limit))
+    value = json.dumps({
+        proxy: {
+            "quota": {
+                "limit": quota,
+                "interval": quota_interval,
+                "timeunit": quota_time_unit,
+                "enabled": quota_enabled
+            },
+            "spikeArrest": {
+                "ratelimit": rate_limit,
+                "enabled": rate_enabled
+            }
+        }
+    })
+
+    rate_limiting = {'ratelimiting': value}
+
+    if (isinstance(apigeeObj, ApigeeApiProducts)):
+        asyncio.run(apigeeObj.update_attributes(rate_limiting))
+    elif isinstance(apigeeObj, ApigeeApiDeveloperApps):
+        asyncio.run(apigeeObj.set_custom_attributes(rate_limiting))
```

Comment:
What happens if I provide an app and product?

Reply:
Might be useful to have a fixture that can update the rate limit of just an app? (Unless there isn't one already)

Reply:
Ah right i've misread this, yeah this is fine